app/src/main/java/com/amaze/filemanager/ui/activities/MainActivity.java

```diff
@@ -521,9 +524,15 @@ private void checkForExternalIntent(Intent intent) {
 
       } else if (actionIntent.equals(Intent.ACTION_SEND) && type != null) {
         // save a single file to filesystem
-        Uri uri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+        Bundle extras = intent.getExtras();
+        String data =
```

Comment:
This will break the existing feature of saving a file from intent extras. Say, you share a file from email attachment, and choose Amaze...
Better null check the `intent.getParcelableExtra(Intent.EXTRA_STREAM)` and then only create data string.

Reply:
I have gone ahead and added a filter to the type of intent and i.e the current code should be executed only when type != text and I am adding a new block for type = text